core/app/assets/js/ui/molecules/buttons-api.scss

```diff
@@ -0,0 +1,124 @@
+$button-secondary-active-background-color: darken($button-secondary-background-color, $button-active-constant);
+$button-secondary-color: theme-colors(light);
+$button-cta-background-color: theme-colors(cta);
+$button-cta-hover-background-color: darken($button-cta-background-color, $button-hover-constant);
+$button-cta-active-background-color: darken($button-cta-background-color, $button-active-constant);
+$button-cta-color: theme-colors(light);
+$button-link-background-color: theme-elements-colors(link-color);
+$button-link-hover-background-color: darken($button-link-background-color, $button-hover-constant);
+$button-link-active-background-color: darken($button-link-background-color, $button-active-constant);
+$button-link-color: theme-colors(light);
+$button-disabled-background-color: theme-colors(disabled);
+$button-disabled-hover-background-color: darken($button-disabled-background-color, $button-hover-constant);
+$button-disabled-active-background-color: darken($button-disabled-background-color, $button-active-constant);
+$button-disabled-color: theme-colors(light);
+// -- dark
+$button-dark-primary-background-color: dark-theme-colors(success);
+$button-dark-primary-color: dark-theme-colors(light);
+$button-dark-primary-hover-background-color: darken($button-dark-primary-background-color, $button-hover-constant);
+$button-dark-primary-active-background-color: darken($button-dark-primary-background-color, $button-active-constant);
+$button-dark-secondary-background-color: dark-tints(400);
+$button-dark-secondary-hover-background-color: darken($button-dark-secondary-background-color, $button-hover-constant);
+$button-dark-secondary-active-background-color: darken($button-dark-secondary-background-color, $button-active-constant);
+$button-dark-secondary-color: dark-theme-colors(light);
+$button-dark-cta-background-color: dark-theme-colors(cta);
+$button-dark-cta-hover-background-color: darken($button-dark-cta-background-color, $button-hover-constant);
+$button-dark-cta-active-background-color: darken($button-dark-cta-background-color, $button-active-constant);
+$button-dark-cta-color: dark-theme-colors(light);
+$button-dark-link-background-color: theme-elements-colors(link-color);
+$button-dark-link-hover-background-color: darken($button-dark-link-background-color, $button-hover-constant);
+$button-dark-link-active-background-color: darken($button-dark-link-background-color, $button-active-constant);
+$button-dark-link-color: theme-colors(light);
+$button-dark-disabled-background-color: dark-theme-colors(disabled);
+$button-dark-disabled-hover-background-color: darken($button-dark-disabled-background-color, $button-hover-constant);
+$button-dark-disabled-active-background-color: darken($button-dark-disabled-background-color, $button-active-constant);
+$button-dark-disabled-color: dark-theme-colors(light);
+
+// using mixing to transfer custom properties
```

Comment:
Why do you need mixins here?

Reply:
@yonatanelm did most of the work here...
It needs more work done on it.